### PR TITLE
packaging: Add Fedora packaging files

### PR DESCRIPTION
**Pull request by Conan-Kudo**
This PR adds the packaging data for Fedora originating from [the Fedora Dist-Git](https://src.fedoraproject.org/cgit/rpms/snapd.git).

While the Fedora packaging in Dist-Git is the authoritative source, a copy of this in this repository will help enable spread testing.

Note the snapd package cannot be built without the patch from #3162.

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
@morphis It's dropped, but the packaging is effectively useless now, since it won't build.

File: packaging/fedora/snap-mgmt.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+# Overlord management of snapd for package manager actions.
+# Implements actions that would be invoked in %pre(un) actions for snapd.
+# Derived from the snapd.postrm scriptlet used in the Ubuntu packaging for
+# snapd.
+
+set -e
+
+SNAP_MOUNT_DIR="/var/lib/snapd/snap"
+SNAP_UNIT_PREFIX="var-lib-snapd-snap"
```
Comment (morphis) [inline packaging/fedora/snap-mgmt.sh]: You can use `$(systemd-escape -p $SNAP_MOUNT_DIR)` here

**Comment by morphis** [comment]
> @morphis It's dropped, but the packaging is effectively useless now, since it won't build.

Thanks! See https://github.com/snapcore/snapd/pull/3365/commits/67673a2c86263a401d078d7c82d2e1a43abfa389 for a fix for the last patch we have in Fedora and not merged upstream yet. Actually the fixed LD script is only available in F26 so we need to fix this from the spec file to have a proper fix available for F24-F26.

File: packaging/fedora-25 [hunk 2]
```diff
@@ -0,0 +1 @@
+fedora
```
Comment (zyga) [inline packaging/fedora-25]: :heart: 

Thank you @Conan-Kudo and @morphis for getting us here :-)

**Comment by zyga** [comment]
I'm merging this. Please iterate on subsequent changes.

**Pull request merged**